src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordToString.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// The record includes a synthesized override of object.ToString().
+    /// For `record R(int I) { public int J; }` it prints `R { I = ..., J = ... }`.
+    /// 
+    /// The method can be declared explicitly. It is an error if the explicit 
+    /// declaration does not match the expected signature or accessibility, or 
+    /// if the explicit declaration doesn't allow overriding it in a derived type and 
+    /// the record type is not sealed. 
+    /// It is an error if either synthesized, or explicitly declared method doesn't 
+    /// override `object.ToString()` (for example, due to shadowing in intermediate base types, etc.).
+    /// </summary>
+    internal sealed class SynthesizedRecordToString : SynthesizedRecordObjectMethod
+    {
+        MethodSymbol _printMethod;
+        public SynthesizedRecordToString(SourceMemberContainerTypeSymbol containingType, MethodSymbol printMethod, int memberOffset, DiagnosticBag diagnostics)
+            : base(containingType, WellKnownMemberNames.ObjectToString, memberOffset, diagnostics)
+        {
+            Debug.Assert(printMethod is object);
```

Comment:
This seems unnecessary since the call site of this constructor has `#nullable enable`.

Reply:
I prefer to keep the assertion until we have `!!` ;-)

---
In reply to: [469418166](https://github.com/dotnet/roslyn/pull/46616#discussion_r469418166) [](ancestors = 469418166)